lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py

```diff
@@ -135,7 +136,10 @@ def main():
                 params['tags'] = ','.join(tags)
             job = tower_cli.get_resource('job')
 
-            lookup_fields = ('job_template', 'inventory', 'credential')
+            lookup_fields = ['job_template', 'inventory']
+            if credential is not None:
+                lookup_fields.append('credential')
```

Comment:
I'm having a hard time seeing how this code will behave any differently. If the credential field is not provided, it looks like it should be `None`, in which case L145 `name = params.pop(field)` will return that and `if name:` resolves to False, meaning that it already isn't added to `params`, which is passed to the launch method.

The only functional difference I would predict is that _after_ the change, `params['credential']` has a value of None, whereas before the key was not present.

Reply:
Yes you are right. Wouldn't behave differently without changes.  Looked in to the history of file and it is fixed in [commit](https://github.com/ansible/ansible/commit/377fba3d766a57957f5b1f56f03e98ecddbb7844#diff-cac7a45d6726902b2eeb5134bc1492f7)

My local ansible fork was not synced and thus was looking at the old code.

Please reject the PR.